django/forms/forms.py

```diff
@@ -492,6 +492,8 @@ def get_initial_for_field(self, field, field_name):
         value = self.initial.get(field_name, field.initial)
         if callable(value):
             value = value()
+        if isinstance(field, MultiValueField) and hasattr(field.widget, 'decompress'):
```

Comment:
Giving `BaseForm` knowledge of specific form fields using `isinstance()` doesn't seem like an appropriate separation of concerns.

Reply:
So maybe we can move that thing to MultiValueField? This will mean that MultiValueField will need to check whether is disabled and if so try to decompress the value before validation.

Reply:
Yes, that sounds better.